compiler/util/misc.cpp

```diff
@@ -73,8 +73,14 @@ bool requireWideReferences() {
   return !fLocal || forceWidePtrs();
 }
 
+//
+// If the --no-local flag is used or the locale model is not 'flat'
+// (i.e., has sub-locales that on-clauses might target), we should
+// not drop on-clauses on the floor.  I.e., we should require them
+// to be "outlined."
+//
 bool requireOutlinedOn() {
-  return !fLocal || forceWidePtrs();
+  return !fLocal || strcmp(CHPL_LOCALE_MODEL, "flat");
```

Comment:
Hoisted by my own petard!  Yes, !=0.

It occurs to me that, in order to increase congruence in the code and comment, you might reverse the end of the block comment above.  Since the name of the function is requireOutlinedOn, maybe say it like this:
    ... we should require on-clauses to be "outlined".  I.e., we should not drop them on the floor.

Sorry I didn't spot that earlier.